### ISSUE TITLE
loaded model gives random predictions in new python session

### ISSUE THREAD

**Issue by lotempeledGong** [fchollet/keras#6634]
Hi, I'm running Keras 2.0.4 with Tensorflow 1.1.0 backend on Python 3.5 .

I'm trying to train and save a sequential model as follows:

```
model = Sequential()
model.add(Embedding(vocab_size, embedding_size, input_length=input_length))
model.add(Bidirectional(GRU(hidden_size, return_sequences=True, activation="tanh")))
model.add(TimeDistributed(Dense(num_labels, activation='softmax')))
optimizer = Adagrad(lr=learning_rate)
model.compile(loss='categorical_crossentropy', optimizer=optimizer, metrics=['categorical_accuracy'])
callbacks = [EarlyStopping(monitor='val_loss', patience=patience, verbose=0)]
model.fit(x_train, y_train, batch_size=batch_size, epochs = epochs, callbacks=callbacks, validation_data=[x_dev, y_dev])

model.save("model.h5")
```

Evaluation at this point gives me accuracy of about 70.

Then I close the python session, open a different one and try:

`model2 = load_model('model.h5')`

Evaluation at this point gives me an accuracy of about 20.

This happens whether I use model.save, model.save_weights, or model.to_json.

**Comment by bfarzin** [comment]
Can you post more code around your `load_model` call?  I use the `to_json` and `load_weights` method and for that case I need to call a `model.compile()` to get it to give me results.  I am less familiar with `model.save()`

**Comment by NikhilKothari** [comment]
Hi guys! I'm facing the same issue. Need some help desperately.
